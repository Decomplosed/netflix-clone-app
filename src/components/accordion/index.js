import React from 'react';

export default function Accordion({ children, ...restProps }) {
  return <Container></Container>;
}
