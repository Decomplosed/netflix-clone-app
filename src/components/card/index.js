import React from 'react';

export const FeatureContext = createContext();
