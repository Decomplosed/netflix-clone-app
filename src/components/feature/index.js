import React from 'react';
import { Container, Title, SubTitle } from './styles/feature';

export default function Feature({ children, ...restProps }) {
  return <Container></Container>
}
