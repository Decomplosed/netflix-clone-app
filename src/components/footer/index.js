import React from 'react';

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}></Container>;
}
