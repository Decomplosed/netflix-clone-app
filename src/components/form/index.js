import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};
