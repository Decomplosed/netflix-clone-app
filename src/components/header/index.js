import React from 'react';

export default Header({bg = true, children, ...restProps}) {}