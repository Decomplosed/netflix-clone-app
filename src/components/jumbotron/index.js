import React from 'react';

export default Jumbotron = () => {
  return <p>I am jumbotron</p>;
};
