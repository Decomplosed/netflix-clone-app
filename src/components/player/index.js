import React, { useState, useContext, createContext } from 'react';
import ReactDOM from 'react-dom';
