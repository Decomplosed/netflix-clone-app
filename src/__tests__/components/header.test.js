import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Header } from '../../components';

jest.mock('react-router-dom');

describe('<Header />', () => {
  it('Renders the basic <Header /> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src='/logo.svg' alt='Netflix' />
          <Header.TextLink active='true'>Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>,
    );

    expect(getByText('Hello I am a link!')).toBeTruthy();
    expect(getByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders the basic <Header /> without a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src='/logo.svg' alt='Netflix' />
          <Header.ButtonLink>Sign In</Header.ButtonLink>
          <Header.TextLink active={false}>Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>,
    );

    expect(getByText('Hello I am a link!')).toBeTruthy();
    expect(queryByTestId('header-bg')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders the full <Header> with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header src='joker1' dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src='/images/logo.svg' alt='Netflix' />
            <Header.TextLink active={false} onClick={() => {}}>
              Series
            </Header.TextLink>
            <Header.TextLink active onClick={() => {}}>
              Films
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search searchTerm='Joker' setSearchTerm={() => {}} />
            <Header.Profile>
              <Header.Picture src='/images/karl.png' />
              <Header.Dropdown>
                <Header.Group></Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
      </Header>,
    );
  });
});
