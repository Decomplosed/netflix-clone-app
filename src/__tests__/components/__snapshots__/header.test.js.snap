// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders the basic <Header /> with a background 1`] = `
<div
  class="header__Background-sc-19jcj5i-0 bzBYTS"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-19jcj5i-1 doewxm"
  >
    <p
      class="header__Link-sc-19jcj5i-2 cdJges"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> Renders the basic <Header /> without a background 1`] = `
<div
  class="header__Container-sc-19jcj5i-1 doewxm"
>
  <p
    class="header__Link-sc-19jcj5i-2 kvZKjY"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header /> Renders the full <Header> with a background 1`] = `
<div
  class="header__Background-sc-19jcj5i-0 bPdakY"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-sc-19jcj5i-1 doewxm"
  >
    <div
      class="header__Group-sc-19jcj5i-3 fZRhLh"
    >
      <p
        class="header__Link-sc-19jcj5i-2 kvZKjY"
      >
        Series
      </p>
      <p
        class="header__Link-sc-19jcj5i-2 kvZKjY"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-sc-19jcj5i-3 fZRhLh"
    >
      <div
        class="header__Search-sc-19jcj5i-12 fqSswf"
      >
        <button
          class="header__SearchIcon-sc-19jcj5i-14 dAmYIn"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-sc-19jcj5i-13 ikBMLR"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-sc-19jcj5i-9 hSgYOx"
      >
        <button
          class="header__Picture-sc-19jcj5i-7 lbwHqG"
          src="/images/users//images/karl.png.png"
        />
        <div
          class="header__Dropdown-sc-19jcj5i-8 bVGkMn"
        >
          <div
            class="header__Group-sc-19jcj5i-3 fZRhLh"
          >
            <button
              class="header__Picture-sc-19jcj5i-7 lbwHqG"
              src="/images/users//images/karl.png.png"
            />
            <p
              class="header__Link-sc-19jcj5i-2 kvZKjY"
            >
              Bart Man
            </p>
          </div>
          <div
            class="header__Group-sc-19jcj5i-3 fZRhLh"
          >
            <p
              class="header__Link-sc-19jcj5i-2 kvZKjY"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-19jcj5i-1 header__Feature-sc-19jcj5i-10 doewxm meFgV"
  >
    <h2
      class="header__FeatureCallOut-sc-19jcj5i-11 jyFuux"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-sc-19jcj5i-4 jTALVQ"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-sc-19jcj5i-15 izSaAL"
    >
      Play
    </button>
  </div>
</div>
`;
